docs/tutorials/adding-a-module-to-your-runtime.md

```diff
@@ -0,0 +1,306 @@
+
+```rust
+// From the reference documentation, but also can be found in the Contract module code
+type Currency: Currency<Self::AccountId>
+```
+
+Fortunately, the Balances module implements this type, so we can simply reference `Balances` to gain access to it.
+
+Similarly, `type DetermineContractAddress` requires the trait `ContractAddressFor`. The Contract module itself implements a type with this trait in `contract::SimpleAddressDeterminator`, thus can use that implementation to satisfy our `contract::Trait`. At this point, I really recommend you explore the source code of the [Contract module](https://github.com/paritytech/substrate/blob/v1.0/srml/contract/src/lib.rs) if things don't make sense or you want to gain a deeper understanding.
+
+### Adding Contract to the Construct Runtime Macro
+
+Next, we need to add the module to the `construct_runtime!` macro.
+
+If we look at the Contract module in detail, we know it has:
+
+* Module **Storage**
+* Module **Event**s
+* **Call**able Functions (dispatchable functions defined in `decl_module!`)
+* Genesis **Config**uration Options
+* (and it is a **Module**)
+
+Thus, when we add the module, it will look like this:
+
+```rust
+construct_runtime!(
+    pub enum Runtime with Log(InternalLog: DigestItem<Hash, AuthorityId, AuthoritySignature>) where
+        Block = Block,
+        NodeBlock = opaque::Block,
+        UncheckedExtrinsic = UncheckedExtrinsic
+    {
+        ...
+        Contract: contract::{Module, Call, Storage, Event<T>, Config<T>},
+    }
+);
+```
+
+### Adding Runtime Hooks
+
+Substrate provides the ability for modules to expose "hooks" where changes in the module can trigger functions in other modules. For example, you could create a module which executes some action every time a new account is created (when it first gains a balance over the [existential deposit](https://substrate.dev/docs/en/overview/glossary#existential-deposit)).
+
+In the case of the Contracts module, we actually want a hook when an account runs out of free balance. Because the Contract module instantiates contracts as "Accounts", it also needs to know when an account is destroyed so that it can clean up any storage that contract was using. You can find that logic in the Contract module source code:
+
+**Contract module `lib.rs`**
+```rust
+impl<T: Trait> OnFreeBalanceZero<T::AccountId> for Module<T> {
+	fn on_free_balance_zero(who: &T::AccountId) {
+		<CodeHashOf<T>>::remove(who);
+		<AccountInfoOf<T>>::get(who).map(|info| child::kill_storage(&info.trie_id));
+	}
+}
+```
+
+To enable this, we simply need to add `Contract` to the `OnFreeBalanceZero` hook provided by the Balances module:
+
+**`contract-chain/runtime/src/lib.rs`**
+```rust
+impl balances::Trait for Runtime {
+    /// What to do if an account's free balance gets zeroed.
+    type OnFreeBalanceZero = (Contract);
```

Comment:
Notice this is the capital C Contract that we defined in construct_runtime.

@shawntabrizi Your discretion on this. I tripped on it for a sec, but it was straightforward to fix.
```
help: a type alias with a similar name exists
    |
184 |     type OnFreeBalanceZero = (Contract);
    |     
```

Reply:
changing to:

```
To enable this, we simply need to add `Contract` type that we defined in the `construct_runtime!` macro to the `OnFreeBalanceZero` hook provided by the Balances module:
```

May be enough to remind the user they need to use the capital Contract